src/frontend/src/binder/select.rs

```diff
@@ -238,6 +244,55 @@ impl Binder {
         Ok((select_list, aliases))
     }
 
+    /// `bind_get_user_by_id_select` binds a select statement that returns a single user name by id,
+    /// this is used for function `pg_catalog.get_user_by_id()`.
+    pub fn bind_get_user_by_id_select(&mut self, input: &ExprImpl) -> Result<BoundSelect> {
+        let select_items = vec![ExprImpl::InputRef(Box::new(InputRef::new(
+            PG_USER_NAME_INDEX,
+            DataType::Varchar,
+        )))];
+        let schema = Schema {
+            fields: vec![Field::with_name(
+                DataType::Varchar,
+                UNNAMED_COLUMN.to_string(),
+            )],
+        };
+        let input = match input {
+            ExprImpl::InputRef(input_ref) => ExprImpl::CorrelatedInputRef(Box::new(
+                CorrelatedInputRef::new(input_ref.index, input_ref.data_type.clone(), 1),
+            )),
+            ExprImpl::CorrelatedInputRef(col_input_ref) => {
+                ExprImpl::CorrelatedInputRef(Box::new(CorrelatedInputRef::new(
+                    col_input_ref.index(),
+                    col_input_ref.return_type(),
+                    col_input_ref.depth() + 1,
+                )))
+            }
+            ExprImpl::Literal(_) => input.clone(),
+            _ => unimplemented!("Unsupported input type"),
```

Comment:
`return Err` rather than panic. The message can also refer to `get_user_by_id` to ease debugging.